### DISCUSSION TITLE
Any interest in a method to convert from PEP 440?

### DISCUSSION THREAD

**Discussion by jamwil** [python-semver/python-semver#390] [Ideas]
I was thinking about making a small library for converting between PEP 440 and SemVer versions. Would a PR for this be in scope?

**Comment by tomschr** [comment]
Hi @jamwil, :wave: 

thank you for your question! Have you looked into our documentation? :wink: You can find about PEP 440 here:

https://python-semver.readthedocs.io/en/latest/advanced/convert-pypi-to-semver.html

It's a start and I'lI assume it's probably incomplete.

How would your "small library" look like? I'm a bit hesitant to add it to the Version class itself. IMHO it should be solely about a semver version. However, we could think about having an adapter class or the function that is shown in the documentation. Or we just extend only the documentation to keep the semver library small.

What do you think, James? Would love to hear your ideas.

**Reply by jamwil** [reply]
Well, jeez. I missed that bit of documentation entirely 🤦‍♂️. Thanks for your thorough reply.

I expect this guide is entirely sufficient for most cases. I agree with you that the library should stick to doing semver well and should avoid feature creep. I'll play with this today and report back if I learn anything valuable (I often don't).

**Reply by tomschr** [reply]
Great, thanks James! I'm glad that the docs is a start. If you find something that can be improved, let me know or create a PR. Happy to improve the docs.

Many thanks!
